File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02683/p02683_num3_parsed.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int N, M, X;
    cin >> N >> M >> X;

    vector<int> C(N);
    vector<vector<int>> A(N, vector<int>(M));

    for(int i = 0; i < N; i++) {
        cin >> C[i];
        for(int j = 0; j < M; j++) {
            cin >> A[i][j];
        }
    }

    int ans = INT_MAX;

    for(int mask = 0; mask < (1 << N); mask++) {
        vector<int> skill(M, 0);
        int cost = 0;

        for(int i = 0; i < N; i++) {
            if(mask & (1 << i)) {
                cost += C[i];
                for(int j = 0; j < M; j++) {
                    skill[j] += A[i][j];
                }
            }
        }

        bool flag = true;
        for(int i = 0; i < M; i++) {
            if(skill[i] < X) {
                flag = false;
                break;
            }
        }

        if(flag) {
            ans = min(ans, cost);
        }
    }

    if(ans == INT_MAX) {
        cout << -1 << endl;
    } else {
        cout << ans << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02683/p02683_num3_parsed.py
def solve(N, M, X, C, A):
    ans = float('inf')

    for mask in range(1 << N):
        cost = 0
        skills = [0] * M

        for i in range(N):
            if (mask >> i) & 1:
                cost += C[i]
                for j in range(M):
                    skills[j] += A[i][j]

        if all(skill >= X for skill in skills):
            ans = min(ans, cost)

    return ans if ans != float('inf') else -1


if __name__ == '__main__':
    N, M, X = map(int, input().split())
    C = []
    A = []

    for _ in range(N):
        temp = list(map(int, input().split()))
        C.append(temp[0])
        A.append(temp[1:])

    print(solve(N, M, X, C, A))
